File: Euclidean_algorithm/PHP/jakobhans/euclidean_test.php
<?php
require "euclidean.php";

class Euclidean extends PHPUnit_Framework_TestCase{
    // Test method gcm_division
    public function test_gcd_division(){
        $this->assertEquals(15, gcd_division(210,45));
        $this->assertEquals(0, gcd_division(0,0));
        $this->assertEquals(1, gcd_division(20,97));
    }

    // Test method gcd_substraction
    public function test_gcd_substraction(){
        $this->assertEquals(15, gcd_substraction(210,45));
        $this->assertEquals(0, gcd_substraction(0,0));
        $this->assertEquals(1, gcd_substraction(20,97));
    }
    // Test method gcd_recursive
    public function test_gcd_recursive(){
        $this->assertEquals(15, gcd_recursive(210,45));
        $this->assertEquals(0, gcd_recursive(0,0));
        $this->assertEquals(1, gcd_recursive(20,97));
    }
}

File: Euclidean_algorithm/Ruby/mitogh/euclidean_test.rb
require './euclidean.rb'

describe "#gcd_divsion" do
  context "Test GCD of 210 and 45" do
    it{ gcd_division(210,45).should eq(15) }
  end
  context "Test GCD of 0 and 0" do
    it{ gcd_division(0,0).should eq(0) }
  end
  context "Test GCD of 20 and 97" do
    it{ gcd_division(20,77).should eq(1) }
  end
end

describe "#gcd_substraction" do
  context "Test GCD of 210 and 45" do
    it{ gcd_substraction(210,45).should eq(15) }
  end
  context "Test GCD of 0 and 0" do
    it{ gcd_substraction(0,0).should eq(0) }
  end
  context "Test GCD of 20 and 97" do
    it{ gcd_substraction(120,77).should eq(1) }
  end
end

describe "#gcd_divsion" do
  context "Test GCD of 210 and 45" do
    it{ gcd_recursive(210,45).should eq(15) }
  end
  context "Test GCD of 0 and 0" do
    it{ gcd_recursive(0,0).should eq(0) }
  end
  context "Test GCD of 20 and 97" do
    it{ gcd_recursive(20,77).should eq(1) }
  end
end
